command/root.go

```diff
@@ -364,24 +367,61 @@ func determineEditor(cmd *cobra.Command) (string, error) {
 }
 
 func ExpandAlias(args []string) ([]string, error) {
-	empty := []string{}
 	if len(args) < 2 {
 		// the command is lacking a subcommand
-		return empty, nil
+		return []string{}, nil
 	}
 
 	ctx := initContext()
 	cfg, err := ctx.Config()
 	if err != nil {
-		return empty, err
+		return nil, err
 	}
 	aliases, err := cfg.Aliases()
 	if err != nil {
-		return empty, err
+		return nil, err
 	}
 
 	expansion, ok := aliases.Get(args[1])
 	if ok {
+		if strings.HasPrefix(expansion, "!") {
+			shellArgs := []string{"-c", expansion[1:]}
+			shellCmd := "sh"
+			if runtime.GOOS == "windows" {
+				shellCmd = "pwsh"
+				argList := ""
+				if len(args[2:]) > 0 {
+					argList = " -ArgumentList @("
+					for i, arg := range args[2:] {
+						argList += fmt.Sprintf("'%s'", arg)
```

Comment:
We've successfully avoided having to shell-escape arguments to `sh` interpreter, but here we're back on square one with PowerShell since arguments with characters such as `'` will cause syntax errors if they are inserted into the above expression.

Is there a way of passing arguments to PowerShell that doesn't require us to escape them? E.g.

```go
exec.Command("pwsh", "-Command", pwshCommand, "-args", args[2:]...)
// if pwshCommand needs to pass arguments internally, it can maybe use `$args`?
```

Reply:
This fails (and was where I started :) ).

Passing arguments like this requires  a literal comma separated list, unlike the `sh` family of interpreters. At a minimum I can join the arguments together with `,`, but this fails for arguments with spaces. It is back where we started with `sh` but unlike `sh` I could find no equivalence to `--`. 

I think that breaking on `'` is bad but an acceptable edge case (at least for now) if we decide to ship pwsh support here.